src/main/java/pl/pelikan/controller/UserController.java

```diff
@@ -0,0 +1,72 @@
+package pl.pelikan.controller;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import pl.pelikan.model.User;
+import pl.pelikan.service.UserService;
+
+import java.security.Principal;
+
+@Controller
+public class UserController {
+
+    private boolean justCreatedAccount;
```

Comment:
pola w controllerach (inne niż inne service/repository) to bardzo niebezpieczna rzecz. Tutaj założyłeś, że z aplikacji korzysta tylko jedna osoba w jednym wątku. W rzeczywistości korzystać z aplikacji korzysta wiele osób i rózne metody w controlerze wywoływane będą w tym samym czasie (równolegle). Jak chcemy mieć już pola w controlerze, to na pewno muszą być synchronizowane (albo atomowe). To są zagadnienia z współbieżności, które na razie nie musisz znać, ale warto zapamiętać, że dla własnego bezpieczeństwa nie chcesz tworzć tu "zwykłch" pól, bo aplikacja na pewno Ci się prędzej czy później wykrzaczy :P

Reply:
dodatkowo przez to wprowadzasz tak zwaną stanowość, czyli już nie możesz mieć parę instancji tej samej aplikacji, bo nie będzie poprawnie działać, co mocno ogranicz Ci skalowanie